lightwood/encoders/time_series/helpers/rnn_helpers.py

```diff
@@ -28,17 +24,24 @@ def tensor_from_series(series, device, n_dims, pad_value, max_len=None, normaliz
             series[i].append(pad_value)
         series[i] = series[i][:max_len]
 
-    # dimension padding
-    for _ in range(max(0, n_dims - len(series))):
-        series.append([pad_value] * max_len)
-
-    # normalize and transpose
     if normalizer:
-        tensor = torch.Tensor([normalizer.encode(s) for s in series][0])
+        if isinstance(normalizer, MinMaxNormalizer):
+            tensor = torch.Tensor([normalizer.encode(s) for s in series]).transpose(0, 1)
+        else:
+            tensor = torch.Tensor([normalizer.encode(s) for s in series][0])
+
+        if len(tensor.shape) > 2:
+            tensor = tensor.squeeze(2)
     else:
-        tensor = torch.transpose(torch.Tensor(series), 0, 1)
+        tensor = torch.Tensor(series).transpose(0, 1)
+
+    # dimension padding
+    for _ in range(max(0, n_dims - tensor.shape[-1])):
+        pad = torch.full((tensor.shape[0], 1), pad_value)
+        tensor = torch.cat([tensor, pad], dim=-1)
```

Comment:
One thing that might be worth noting on here is if the padding is being done "forward" or "backwards".

It should be done "backwards" I think, i.e. [1,2,3] where the max length is `5` then we have [0,0,1,2,3] because the bigger and index is the more recent the value, and usually (always?) when padding is needed it's older data that's missing.

Reply:
You are right, it should be backwards